### PR TITLE
Added autodetection for force-feedback devices.

### PR DESCRIPTION
**Pull request by peci1**
The detection algorithm is best described by inline code comments. Basically it searches for a relevant event device in `/dev/input/by-id`.

One thing for consideration: do we want this autodetection to run every time the `dev_ff` parameter is empty? Or should there be a magical value like `auto` that triggers the autodetection?

### CODE REVIEW

**Comment by peci1** [comment]
The autodetection is run every time a new joy device is open, so if you have a F710 and specify `_dev:=/dev/input/js0`, you can play with the X-D toggle and the node always keeps track of the correct feedback device.

**Comment by JWhitleyWork** [comment]
@peci1 I'm very sorry that it took so long to get to this PR. Can you please rebase on the `main` branch (changed from `master`)? I'd like to have Github Actions run over this. I'll also take a look.

File: joy/src/joy_node.cpp [hunk 1]
```diff
@@ -162,6 +163,96 @@ class Joystick
...
+      const std::string dev_real(realpath_buf);
+      if (dev_real == joy_dev_real)
+      {
+        // we found the ID device which maps to the selected joy
+        joy_dev_id = current_path;
+        break;
+      }
+    }
+
+    closedir(dev_dir);
```
Comment (JWhitleyWork) [inline joy/src/joy_node.cpp:221]: To make sure there are no memory leaks, can you set up an RAII class for this with the `closedir` in the destructor?

File: joy/src/joy_node.cpp [hunk 2]
```diff
@@ -162,6 +163,96 @@ class Joystick
...
+
+      const auto current_path = std::string(path) + "/" + entry->d_name;
+      if (current_path.find(joy_dev_id_prefix) != std::string::npos)
+      {
+        ROS_INFO("Found force feedback event device %s", current_path.c_str());
+        event_dev = current_path;
+        break;
+      }
+    }
+    closedir(dev_dir);
```
Comment (JWhitleyWork) [inline joy/src/joy_node.cpp:251]: Same here.

**Comment by peci1** [comment]
Thanks for taking a look. I pushed an attampt at RAII.

**Comment by JWhitleyWork** [comment]
@peci1 Thanks! Would you mind switching it to a `uniute_ptr` instead since there will be no change of ownership of the pointer? After that, I'm good to merge it.

**Comment by peci1** [comment]
Okay, I've changed to unique_ptr (although the syntax is a little bit more complicated).

**Comment by peci1** [comment]
@JWhitleyWork Would you please have a look at backporting this to melodic? Or should I prepare a PR?

**Pull request merged**